common/libs/framing/frame_detector.h

```diff
@@ -0,0 +1,129 @@
+#ifndef __FRAME_DETECTOR__
+#define __FRAME_DETECTOR__
+
+#include "serial_listeners.h"
+#include <stdint.h>
+#include <string.h>
+
+template <class RxBuffer, int FRAME_BUF_LEN>
+class FrameDetector : public RxListener {
+public:
+  enum class State {
+    LOST,       // Frame detector does not know where it is in the stream of
+                // bytes coming into RxBuffer
+    WAIT_START, // Frame detector for a frame start marker
```

Comment:
Frame detector **is waiting** for a frame start marker?

Also how about renaming the states: WAIT_START -> WAITING_FOR_START_MARKER, RX_FRAME -> RECEIVING_FRAME?